Makefile

```diff
@@ -139,50 +132,41 @@ PROVE_PATTERN = "{t,examples}/**/*.t"
 prove: HARNESS ?= TAP::Harness
 prove: PROVE_FILES ?= $(call circleci, $(PROVE_PATTERN))
 prove: export TEST_NGINX_RANDOMIZE=1
-prove: $(ROVER) lua_modules nginx ## Test nginx
+prove: $(ROVER) dependencies nginx ## Test nginx
 	$(ROVER) exec script/prove --verbose -j$(NPROC) --harness=$(HARNESS) $(PROVE_FILES)
 
-prove-docker: apicast-source
-prove-docker: export IMAGE_NAME ?= apicast-test
+prove-docker: export IMAGE_NAME ?= $(DEVEL_IMAGE)
 prove-docker: ## Test nginx inside docker
+	make -C $(PROJECT_PATH) -f $(MKFILE_PATH) apicast-source
 	$(DOCKER_COMPOSE) run --rm -T prove | awk '/Result: NOTESTS/ { print "FAIL: NOTESTS"; print; exit 1 }; { print }'
 
-builder-image: PULL_POLICY ?= always
-builder-image: ## Build builder image
-	$(S2I) build . $(BUILDER_IMAGE) $(IMAGE_NAME) \
-		--context-dir=$(S2I_CONTEXT) \
-		--pull-policy=$(PULL_POLICY) \
-		--runtime-pull-policy=$(PULL_POLICY) \
-		--incremental $(S2I_OPTIONS)
-
-runtime-image: PULL_POLICY ?= always
-runtime-image: IMAGE_NAME = apicast-runtime-image
+runtime-image: IMAGE_NAME ?= apicast-runtime-image:latest
 runtime-image: ## Build runtime image
-	$(S2I) build . $(BUILDER_IMAGE) $(IMAGE_NAME) \
-		--context-dir=$(S2I_CONTEXT) \
-		--runtime-image=$(RUNTIME_IMAGE) \
-		--pull-policy=$(PULL_POLICY) \
-		--runtime-pull-policy=$(PULL_POLICY) \
-		$(S2I_OPTIONS)
+	$(DOCKER) build -t $(IMAGE_NAME) $(PROJECT_PATH)
 
 push: ## Push image to the registry
 	docker tag $(IMAGE_NAME) $(REGISTRY)/$(IMAGE_NAME)
 	docker push $(REGISTRY)/$(IMAGE_NAME)
 
-bash: export IMAGE_NAME ?= apicast-test
+bash: export IMAGE_NAME ?= $(RUNTIME_IMAGE)
 bash: export SERVICE = gateway
-bash: builder-image apicast-source ## Run bash inside the builder image
+bash: ## Run bash inside the runtime image
+	make -C $(PROJECT_PATH) -f $(MKFILE_PATH) apicast-source
 	$(DOCKER_COMPOSE) run --user=root --rm --entrypoint=bash $(SERVICE)
 
-dev: export IMAGE_NAME ?= apicast-test
-dev: export SERVICE = dev
-dev: USER = root
-dev: builder-image apicast-source ## Run APIcast inside the container mounted to local volume
-	$(DOCKER_COMPOSE) run --user=$(USER) --service-ports --rm --entrypoint=bash $(SERVICE) -i
+gateway-logs: export IMAGE_NAME = does-not-matter
+gateway-logs:
+	$(DOCKER_COMPOSE) logs gateway
 
-test-builder-image: export IMAGE_NAME ?= apicast-test
-test-builder-image: clean-containers ## Smoke test the builder image. Pass any docker image in IMAGE_NAME parameter.
+test-runtime-image: export IMAGE_NAME ?= $(RUNTIME_IMAGE)
```

Comment:
Do we only want to test the runtime image or also the development image? I guess only dev image makes sense but just wanted to check this with you first.

Reply:
Related to the previous [comment](https://github.com/3scale/APIcast/pull/1344#discussion_r883490073), I think that only the runtime image should be tests. It does not make sense to test the devel image, or it has low value doing it. wdyt?